code/go/example.go

```diff
@@ -1,68 +1,49 @@
-// go mod init main
+// go mod init my-module-path
 // go run example.go
 package main
 
 import (
 	"fmt"
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
-	"io"
-	"reflect"
 )
 
 func main() {
-	results, err := runQuery("neo4j+s://demo.neo4jlabs.com:7687", "movies", "mUser", "s3cr3t")
+	credentials := neo4j.BasicAuth("mUser", "s3cr3t", "")
+	driver, err := neo4j.NewDriver("neo4j+s://demo.neo4jlabs.com:7687", credentials)
 	if err != nil {
-		panic(err)
+		panic("Could not create driver")
 	}
-	for _, result := range results {
-		fmt.Println(result)
+	defer driver.Close()
+	results, err := runQuery(driver, getDairyProducts)
```

Comment:
we should not have a domain specific function name here

Reply:
wouldn't an inline func not be better?

Reply:
`runQuery` is not domain-specific though, `getDairyProducts` is.
Happy to inline `runQuery`

Reply:
should be renamed getMovies (or similar) as we are now using the movie Db.